File: app/src/main/java/com/mio4kon/kotlin/practice/compareJava/MainActivity.java
package com.mio4kon.kotlin.practice.compareJava;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.mio4kon.kotlin.practice.R;
import com.mio4kon.kotlin.practice.ui.activity.RecyclerViewActivity;

/**
 * Created by mio4kon on 16/8/12.
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Intent intent = new Intent(this,RecyclerViewActivity.class);
       this.startActivity(intent);

    }
}

File: app/src/main/java/com/mio4kon/kotlin/practice/ui/activity/MainActivity.kt
package com.mio4kon.kotlin.practice.ui.activity

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.mio4kon.kotlin.practice.R
import com.mio4kon.kotlin.practice.util.consume
import com.mio4kon.kotlin.practice.util.gotoActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        recyleViewsBt.setOnClickListener {
            gotoActivity<RecyclerViewActivity>()
        }
        gankIoBt.setOnClickListener {
            gotoActivity<GankIoActivity>()
        }

        spBt.setOnClickListener {
            gotoActivity<SpActivity>()
        }

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean = with(menuInflater) {
        inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }


    override fun onOptionsItemSelected(item: MenuItem): Boolean = when(item.itemId) {
        R.id.action_unit->consume{gotoActivity<UnitTestActivity>()}
        else -> super.onOptionsItemSelected(item)
    }


}
